vnext/Microsoft.ReactNative/Modules/NativeUIManager.cpp

```diff
@@ -290,10 +290,13 @@ static YGValue YGValueOrDefault(const winrt::Microsoft::ReactNative::JSValue &va
       folly::dynamic pct(str);
       return YGValue{static_cast<float>(pct.asDouble()), YGUnitPercent};
     }
+    if (str.length() > 2 && (str.substr((str.length() - 2), 2) == "px" || str.substr((str.length() - 2), 2) == "pt")) {
```

Comment:
This is super tricky, but ends up allocating some extra strings that we don't need. Because this function is called frequently, we likely want to avoid this.

Could you instead use [`string::compare`](https://en.cppreference.com/w/cpp/string/basic_string/compare) to compare at a given index without creating a substring?

Reply:
Yes definitely. I'll change that now.